### PR TITLE
137 filter ui

### PR DESCRIPTION
**Pull request by jitendraharpalani58**
See #137 

@danielbachhuber 

Please review that I am on right track. Filter/Search is working with only one problem. Its appending new shortcodes. So, for some reasons my `this.views.unset()` is not working. 

Once you verify, I will add appropriate label for search and also improve a bit of UI for search. Right now, its just an input text box.

### CODE REVIEW

File: js/views/insert-shortcode-list.js [hunk 1]
```diff
@@ -7,10 +8,21 @@ var insertShortcodeList = wp.Backbone.View.extend({
...
+	
+	refresh: function( shortcodeData ) {
+		var options = { shortcodes: new Shortcodes( shortcodeData ) };
+		this.displayShortcodes( options );
+	},
+	
+	displayShortcodes: function(options) {
+		var t = this;
+		
+		t.views.unset();
```
Comment (jitendraharpalani58) [inline js/views/insert-shortcode-list.js]: Not working for unknown reasons.

File: js/controllers/media-controller.js [hunk 2]
```diff
@@ -19,6 +21,15 @@ var MediaController = wp.media.controller.State.extend({
 				this.frame.toolbar.get().refresh();
 			}
 		},
+		
+		search: function( searchTerm ) {
+			var pattern = new RegExp( searchTerm, "gi" );
+			var filteredModels = sui.shortcodes.filter( function( model ) {
+				return pattern.test( model.get( "label" ) );
+            });
```
Comment (danielbachhuber) [inline js/controllers/media-controller.js:29]: Indentation

**Comment by danielbachhuber** [comment]
> Please review that I am on right track.

Looks pretty good!

> Filter/Search is working with only one problem. Its appending new shortcodes.

Bug confirmed :)

**Comment by bfintal** [comment]
Aside from the appearance of the UI, shortcodes aren't being hidden when I tried this out. New stuff are only appended to the list.

**Comment by jitendraharpalani58** [comment]
@danielbachhuber this should be ready to go. 

Please review.

File: js/views/search-shortcode.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	className: 'search',
+	id:        'media-search-input',
+	
+	initialize: function( options ) {
+		this.shortcodeList = options.shortcodeList;
+	}, 
+
+	attributes: {
+		type:        'search',
+		placeholder: 'search'
```
Comment (danielbachhuber) [inline js/views/search-shortcode.js]: This placeholder needs to be translatable.

**Comment by danielbachhuber** [comment]
There's still a bug in this. If I delete text from the search input, then all of the shortcodes should appear again.

**Comment by danielbachhuber** [comment]
:+1: nice work on this @jitendraharpalani58

**Comment by bfintal** [comment]
Works perfectly

**Pull request merged**